src/esp/core/ManagedContainer.h

```diff
@@ -20,8 +20,11 @@ namespace core {
  * @tparam T the type of managed object a particular specialization of
  * this class works with.  Must inherit from @ref
  * esp::core::AbstractManagedObject.
+ * @tparam AccessViaCopies Whether the default access (getters) for this
+ * container provides copies of the objects held, or the actual objects
+ * themselves.
  */
-template <class T>
+template <class T, bool AccessViaCopies>
```

Comment:
The reason why I implemented this change via a non-type template argument instead of a variable passed via the constructor is two fold :

1. By making it a template argument, it changes the actual type.  A ManagedContainer<ObjectAttributes, true> is not the same type as a ManagedContainer<ObjectAttributes, false> and so they cannot inherit or otherwise interfere with each other.
2. The impact of the change was minimal (much less code changed than if it was a constructor argument).

Added Bonus : it exposed a [bug in our ESP_SMART_POINTERS macro](https://github.com/facebookresearch/habitat-sim/pull/1071).  🎉

Reply:
Ah, just wanted to make sure we weren't excessively copying.